rt_gene/src/rt_gene/gaze_estimation_models_pytorch.py

```diff
@@ -1,177 +1,150 @@
-            param.requires_grad = True
-
-        _num_ftrs = 1280 + 1280 + 2  # left, right and head_pose
-        self.classifier = GazeEstimationAbstractModel._create_fc_layers(num_out=num_out, num_features=_num_ftrs)
-        GazeEstimationAbstractModel._init_weights(self.modules())
-
-    def forward(self, left_eye, right_eye, headpose):
-        left_x = self.left_features(left_eye)
-        left_x = left_x.mean([2, 3])
-
-        right_x = self.right_features(right_eye)
-        right_x = right_x.mean([2, 3])
-
-        concat = torch.cat((left_x, right_x, headpose), dim=1)
-
-        fc_output = self.classifier(concat)
-
-        return fc_output
-
-
-class GazeEstimationModelResnet50(GazeEstimationAbstractModel):
-
-    def __init__(self, num_out=2):  # phi, theta
-        super(GazeEstimationModelResnet50, self).__init__()
-        _left_model = models.resnet50(pretrained=True)
-        _right_model = models.resnet50(pretrained=True)
-
-        # remove the last ConvBRelu layer
-        self.left_features = nn.Sequential(
-            _left_model.conv1,
-            _left_model.bn1,
-            _left_model.relu,
-            _left_model.maxpool,
-            _left_model.layer1,
-            _left_model.layer2,
-            _left_model.layer3,
-            _left_model.layer4,
-            _left_model.avgpool
+        concat = nn.Sequential(
+            nn.Linear(2048, 512),
+            nn.BatchNorm1d(512, momentum=0.99, eps=1e-3),
+            nn.ReLU(inplace=True)
         )
 
-        self.right_features = nn.Sequential(
-            _right_model.conv1,
-            _right_model.bn1,
-            _right_model.relu,
-            _right_model.maxpool,
-            _right_model.layer1,
-            _right_model.layer2,
-            _right_model.layer3,
-            _right_model.layer4,
-            _right_model.avgpool
+        fc = nn.Sequential(
+            nn.Linear(514, 256),
+            nn.ReLU(inplace=True),
+            nn.Linear(256, out_features)
         )
 
-        for param in self.left_features.parameters():
-            param.requires_grad = True
-        for param in self.right_features.parameters():
-            param.requires_grad = True
-
-        _num_ftrs = _left_model.fc.in_features + _right_model.fc.in_features + 2  # left, right and head_pose
-        self.classifier = GazeEstimationAbstractModel._create_fc_layers(num_out=num_out, num_features=_num_ftrs)
-        GazeEstimationAbstractModel._init_weights(self.modules())
-
-    def forward(self, left_eye, right_eye, headpose):
-        left_x = self.left_features(left_eye)
-        left_x = torch.flatten(left_x, 1)
-
-        right_x = self.right_features(right_eye)
-        right_x = torch.flatten(right_x, 1)
+        return x_l, x_r, concat, fc
 
-        concat = torch.cat((left_x, right_x, headpose), dim=1)
-
-        fc_output = self.classifier(concat)
-
-        return fc_output
+    @staticmethod
+    def _init_weights(modules):
```

Comment:
Above it's called `initialize_weights` - shall we use the same method names?

Reply:
sure....I'll refactor

Reply:
Done.